libcontainer/cgroups/v1_utils.go

```diff
@@ -90,6 +94,16 @@ func tryDefaultPath(cgroupPath, subsystem string) string {
 	return path
 }
 
+func ReadCgroupMountinfo() ([]*mountinfo.Info, error) {
+	readMountinfoOnce.Do(func() {
```

Comment:
Here we assume that no new cgroup mounts appear after runc (or anyone who uses libcontainer) start, iow cgroup mounts are performed early during system start, earlier than any container orchestration daemons are started.

Reply:
Going through this PR again; perhaps put that information in a GoDoc for the function (I notice various exported functions here are not documented; should we have CI checking for that?

(I left a suggestion above)

Reply:
This is a fairly reasonable assumption to make, and to be honest the previous code didn't handle cgroup mounts changing in a race-free way either (not to mention that the median lifetime of a runc process is measured in milliseconds) so practically speaking this semantic change doesn't matter.